src/gleam/string.gleam

```diff
@@ -517,22 +523,32 @@ pub fn pad_right(
   string: String,
   to desired_length: Int,
   with pad_string: String,
-) {
+) -> String {
   let current_length = length(string)
   let to_pad_length = desired_length - current_length
-  iterator.single(string)
-  |> iterator.append(padding(to_pad_length, pad_string))
-  |> iterator.to_list
-  |> concat
+
+  case to_pad_length <= 0 {
+    True -> string
+    False ->
+      string
+      <> case length(pad_string) {
+        1 -> repeat(pad_string, to_pad_length)
+        pad_string_length ->
+          padding(to_pad_length, pad_string, pad_string_length)
+      }
+  }
```

Comment:
I'd do the same I described for `pad_left`

Reply:
Done